src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -189,8 +189,41 @@ void GcsServer::DoStart(const GcsInitData &gcs_init_data) {
       /*ms*/ RayConfig::instance().event_stats_print_interval_ms(),
       "GCSServer.deadline_timer.debug_state_event_stats_print");
 
+  global_gc_throttler_ =
+      std::make_unique<Throttler>(RayConfig::instance().global_gc_min_interval_s() * 1e9);
+
   periodical_runner_.RunFnPeriodically(
-      [this] { DumpDebugStateToFile(); },
+      [this] {
+        DumpDebugStateToFile();
+
+        if (cluster_task_manager_->GetPendingQueueSize() == 0) {
+          task_pending_schedule_detected_ = 0;
+          return;
+        }
+        // Trigger global gc to solve task pending.
+        // To avoid spurious triggers, only those after two consecutive
+        // detections and under throttling are sent out.
+        if (task_pending_schedule_detected_++ > 0 && global_gc_throttler_->AbleToRun()) {
```

Comment:
It should be `++task_pending_schedule_detected_ > 0`?

Reply:
`++task_pending_schedule_detected_ > 0` would trigger the global gc the first time finding pending tasks in gcs. That might be overly reactive.